src/Crocodealer/Command/Sync.hs

```diff
@@ -0,0 +1,81 @@
+module Crocodealer.Command.Sync
+       ( sync
+       ) where
+
+import Network.HTTP.Simple (Response, getResponseBody, getResponseStatus, httpBS, parseRequest)
+import Network.HTTP.Types.Status (Status (..))
+
+import Crocodealer.Command.Colors (errorMessage, infoMessage, warningMessage)
+
+data File = File
+    { fileName     :: !Text
+    , fileContents :: !Text
+    }
+
+data FileStatus
+    = UpToDate
+    | Outdated
+    | FileDoesNotExist
+
+data Repo = Repo
+    { repoUser   :: !Text
+    , repoName   :: !Text
+    , repoStatus :: !FileStatus
+    }
+
+mkRepo :: Text -> Text -> FileStatus -> Repo
+mkRepo user name status =
+    Repo { repoUser = user, repoName = name, repoStatus = status }
+
+sync :: Text -> Text -> Text -> [Text] -> IO ()
+sync user repoName fileName repositories = do
+    resp <- fetchFile user repoName fileName
+    let file = File { fileName = fileName, fileContents = decodeBody resp }
+
+    case getResponseStatus resp of
+      Status 200 _ -> do
+          updatedRepos <- checkRepositoriesToUpdate user repositories file
+          putTextLn fileName
+          printColors updatedRepos
+      _ -> errorMessage "File not found in template repository!"
+
+fetchFile :: Text -> Text -> Text -> IO (Response ByteString)
+fetchFile user repoName fileName = do
+    let url = "https://raw.githubusercontent.com/" <> user <> "/" <> repoName <> "/master/" <> fileName
+    req <- parseRequest $ toString url
+    httpBS req
+
+printColors :: [Repo] -> IO ()
+printColors = mapM_ colorStatus
+  where
+    colorStatus :: Repo -> IO ()
+    colorStatus status =
+        case status of
+            Repo user repoName UpToDate ->
+                infoMessage $ user <> "/" <> repoName <> ": " <> "Up-to-date"
+            Repo user repoName Outdated ->
+                warningMessage $ user <> "/" <> repoName <> ": " <> "Outdated"
+            Repo user repoName FileDoesNotExist ->
+                errorMessage $ user <> "/" <> repoName <> ": " <> "Doesn't have file"
+
+
+checkRepositoriesToUpdate :: Text -> [Text] -> File -> IO [Repo]
+checkRepositoriesToUpdate user repos file =
+    mapM (checkSingleRepository user file) repos
+
+checkSingleRepository :: Text -> File -> Text -> IO Repo
+checkSingleRepository user file repoName = do
+    resp <- fetchFile user repoName $ fileName file
+
+    case getResponseStatus resp of
+      Status 200 _ ->
+        if checkSame (fileContents file) $ decodeBody resp then
+          pure $ mkRepo user repoName UpToDate else
+          pure $ mkRepo user repoName Outdated
+      _ -> pure $ mkRepo user repoName FileDoesNotExist
+
+checkSame :: Text -> Text -> Bool
+checkSame orgFile repoFile = orgFile == repoFile
```

Comment:
It's great that you've separated pure and impure logic in separate functions! Good stuff :slightly_smiling_face: